src/classes/time.rs

```diff
@@ -119,8 +119,10 @@ impl Time {
         let hour = s.digits00_23()?;
         let (minute, second, subsec) = match s.digits00_59() {
             Some(m) => {
-                let (sec, sub) = match s.digits00_59() {
+                let (sec, sub) = match s.digits00_60_leap() {
                     Some(n) => (n, s.subsec().unwrap_or(SubSecNanos::MIN)),
+                    // If there are more digits but they're not valid seconds, fail the parse
```

Comment:
This shouldn't be needed. The `read_` family of functions will parse as much as possible, leaving the caller to continue scanning and potentially raising if stuff remains.

Was this branch needed to fix something else?